src/CMake/VisItMacros.cmake

```diff
@@ -316,3 +316,82 @@ macro(visit_add_library)
     unset(${val_NAME}_FEATURES CACHE)
 endmacro()
 
+##############################################################################
+# Like visit_add_library, but adds parallel compile/link options.
+# Taken mostly from VISIT_ADD_PARALLEL_LIBRARY
+#
+# ARGUMENTS:
+#    NAME         target name                REQUIRED
+#    SOURCES      [source1 [source2 ...]]    REQUIRED
+#    HEADERS      [header1 [header2 ...]]    OPTIONAL
+#    INCLUDES     [dir1 [dir2 ...]]          OPTIONAL
+#    DEFINES      [define1 [define2 ...]]    OPTIONAL
+#    DEPENDS_ON   [dep1 ...]                 OPTIONAL
+#    FOLDER       [name])                    OPTIONAL
+#
+##############################################################################
+
+macro(visit_add_parallel_library)
+
+    visit_add_library(${ARGV})
+
+    cmake_parse_arguments(vapl "" "NAME" "" ${ARGN})
+
+    if(UNIX)
+        if(VISIT_PARALLEL_CXXFLAGS)
```

Comment:
I am sure I don't understand CMake at sufficient level but something seems odd to me...we're in a function that is intended for handling parallel libraries. Yet, we are still testing for existence of flags related to parallel?

Wondering if this logic makes for easier reading...

```

if (NOT VISIT_PARALLEL_CXXFLAGS)
    message(FATAL_ERROR "Asked to add a parallel library but VISIT_PARALLEL_CXXFLAGS is not set)
endif()

if(UNIX)
    set(PAR_COMPILE_FLAGS "")
    set(PAR_INCLUDE "")
    .
    .
    .
else()
    .
    .
    .
endif()
```

Reply:
VISIT_PARALLEL_CXX_FLAGS is not always defined, and it is not an error if it isn't.